analyzers/PapasFromFccsw.py

```diff
@@ -29,30 +30,48 @@ class PapasEventFromRoot(Analyzer):
     '''
 
     def __init__(self, *args, **kwargs):
-        super(PapasEventFromRoot, self).__init__(*args, **kwargs)
+        super(PapasFromFccsw, self).__init__(*args, **kwargs)
 
     def process(self, event):
         event.papasevent = PapasEvent(event.iEv)
         papasevent = event.papasevent
 
         #make a dict from the gen_particles list so that it can be stored into the papasevent collections
         gen_particles = getattr(event, self.cfg_ana.gen_particles)
-        gen_particles_collection = {x.uniqueid():x for x in gen_particles}
+        for g in gen_particles:
+            #set the papas identifiers for use in DAG
+            g.set_dagid(Identifier.make_id(Identifier.PFOBJECTTYPE.PARTICLE, g.objid()[0], 'g', g.p4().E()))
+        #make a dict from the gen_particles list so that it can be stored into the papasevent collections           
+        gen_particles_collection = {x.dagid():x for x in gen_particles}
 
         #make a dict from the rec_particles list so that it can be stored into the papasevent collections
         rec_particles = getattr(event, self.cfg_ana.rec_particles)
+        for r in rec_particles:
+            #set the papas identifiers for use in DAG
+            r.set_dagid(Identifier.make_id(Identifier.PFOBJECTTYPE.PARTICLE, r.objid()[0], 'r', r.p4().E()))
+                
         #if there are no rec_particles we assume this was an evernt discarded during reconstruction and skip it
         if len(rec_particles) == 0:
             self.mainLogger.error('no reconsrtucted particles found -> Event discarded')
             return False
-        rec_particles_collection = {x.uniqueid():x for x in rec_particles}
+        #make a dict from the rec_particles list so that it can be stored into the papasevent collections           
+        rec_particles_collection = {x.dagid():x for x in rec_particles}
         #create the history links for relationship between gen and rec particles
         particle_links = getattr(event, self.cfg_ana.gen_rec_links)
         for plink in particle_links:
-            nodeid = plink.childid()
-            child = papasevent.history.setdefault(nodeid, Node(nodeid)) #creates a new node if it is not there already
-            nodeid =  plink.parentid()
-            parent = papasevent.history.setdefault(nodeid, Node(nodeid))
+            genid = None
+            recid = None
+            for g in gen_particles:
+                if g.objid() == plink.id1() :
+                    genid = g.dagid()
+                    break
+            for g in rec_particles:
+                if g.objid() == plink.id2() :
+                    recid = g.dagid()
+                    break            
+            #todo add in a throw incase soemthing is not found
+            child = papasevent.history.setdefault(recid, Node(recid)) #creates a new node if it is not there already
+            parent = papasevent.history.setdefault(genid, Node(genid))
```

Comment:
Please see below, line 81: why resetting papasevent.iEv? I assume it's done by the PapasEvent constructor at the beginning

Reply:
you are right